apps/ledger-live-mobile/src/families/ethereum/EditTransaction/MethodSelection.tsx

```diff
@@ -0,0 +1,236 @@
+  ) as Transaction;
+
+  const { transaction, setTransaction, status } =
+    useBridgeTransaction<Transaction>(() => {
+      return {
+        account,
+        parentAccount,
+        transaction: transactionToEdit,
+      };
+    });
+
+  invariant(
+    transaction,
+    "[useBridgeTransaction - MethodSelection] could not found transaction from bridge.",
+  );
+
+  const mainAccount = getMainAccount(account, parentAccount);
+  const feePerGas = new BigNumber(
+    EIP1559ShouldBeUsed(mainAccount.currency)
+      ? transactionToEdit.maxFeePerGas!
+      : transactionToEdit.gasPrice!,
+  );
+
+  const feeValue = new BigNumber(
+    transactionToEdit.userGasLimit || transactionToEdit.estimatedGasLimit || 0,
+  )
+    .times(feePerGas)
+    .div(new BigNumber(10).pow(mainAccount.unit.magnitude));
+
+  const haveFundToCancel = mainAccount.balance.gt(feeValue.times(1.3));
+  const haveFundToSpeedup = mainAccount.balance.gt(
+    feeValue
+      .times(1.1)
+      .plus(account.type === "Account" ? transactionToEdit.amount : 0),
+  );
+
+  const oldestOperation = isOldestEditableOperation(operation, account);
+
+  const currency = getAccountCurrency(account);
+
+  const bridge = getAccountBridge(account, parentAccount as Account);
+
+  const onSelect = useCallback(
+    (option: "cancel" | "speedup") => {
+      switch (option) {
+        case "cancel":
+          {
+            const updatedTransaction: Partial<Transaction> = {
+              amount: new BigNumber(0),
+              data: Buffer.from(""),
+              nonce: operation.transactionSequenceNumber,
+              allowZeroAmount: true,
+              mode: "send",
+            };
+
+            if (EIP1559ShouldBeUsed(mainAccount.currency)) {
+              if (transactionToEdit.maxPriorityFeePerGas) {
+                updatedTransaction.maxPriorityFeePerGas = new BigNumber(
+                  transactionToEdit.maxPriorityFeePerGas.times(1.1),
+                );
+              }
+
+              if (transactionToEdit.maxFeePerGas) {
+                updatedTransaction.maxFeePerGas = new BigNumber(
+                  transactionToEdit.maxFeePerGas.times(1.3),
+                );
+              }
+            } else if (transactionToEdit.gasPrice) {
+              updatedTransaction.gasPrice = new BigNumber(
+                transactionToEdit.gasPrice.times(1.3),
+              );
+            }
+
+            setTransaction(
+              bridge.updateTransaction(transaction, updatedTransaction),
+            );
+          }
+
+          break;
+
+        case "speedup":
+          {
+            const updatedTransaction: Partial<Transaction> = {
+              nonce: operation.transactionSequenceNumber,
+              networkInfo: null,
+              gasPrice: null,
+              maxFeePerGas: null,
+              maxPriorityFeePerGas: null,
+              mode: "send",
+              recipient:
+                (account as Account)?.freshAddress ??
```

Comment:
I think it is not correct here.
```recipient: (account as Account)?.freshAddress```
should be used by cancel flow instead of speedup flow

otherwise, when you speed up a transaction, it will send funds to yourself :(